scripts/deploy/approve-production-release.mjs

```diff
@@ -0,0 +1,25 @@
+import {createAppAuth} from '@octokit/auth-app';
+import {Octokit} from 'octokit';
+
+const authSecrets = {
+  appId: process.env.RELEASER_APP_ID,
+  privateKey: process.env.RELEASER_PRIVATE_KEY,
+  clientId: process.env.RELEASER_CLIENT_ID,
+  clientSecret: process.env.RELEASER_CLIENT_SECRET,
+  installationId: process.env.RELEASER_INSTALLATION_ID,
+};
+
+const octokit = new Octokit({
+  authStrategy: createAppAuth,
+  auth: authSecrets,
+});
+await octokit.request(
```

Comment:
Related to my above question: Can you go into more details what this "approves" exactly ?

How does this makes it so we only publish officially on npm when this script runs ?

Reply:
This is a new thing I added to our app, the ability to approve GitHub environment usage automatically (properly-ish), using https://docs.github.com/en/actions/deployment/protecting-deployments/creating-custom-deployment-protection-rules.

The only quirk here is that instead of relying on webhook to have all the info to approve the run, we provide it when calling the deployment pipeline.